thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/algorithm/DimensionWrapper.java

```diff
@@ -198,21 +200,47 @@ public DetectionPipelineResult run() throws Exception {
     List<MergedAnomalyResultDTO> anomalies = new ArrayList<>();
     Map<String, Object> diagnostics = new HashMap<>();
     Set<Long> lastTimeStamps = new HashSet<>();
-    LOG.info("exploring {} metrics", nestedMetrics.size());
-    for (MetricEntity metric : nestedMetrics) {
-      for (Map<String, Object> properties : this.nestedProperties) {
-        LOG.info("running detection for {}", metric.toString());
-        DetectionPipelineResult intermediate = this.runNested(metric, properties);
-        lastTimeStamps.add(intermediate.getLastTimestamp());
-        anomalies.addAll(intermediate.getAnomalies());
-        diagnostics.put(metric.getUrn(), intermediate.getDiagnostics());
+    long totalNestedMetrics = nestedMetrics.size();
+    long successNestedMetrics = 0; // record the number of successfully explored dimensions
+    LOG.info("exploring {} metrics", totalNestedMetrics);
+    for (int i = 0; i < totalNestedMetrics; i++) {
+      if (i == EARLY_STOP_THRESHOLD && successNestedMetrics == 0) {
```

Comment:
It would be better to put it into a separate method as well.

Reply:
Sure